packages/bff/src/services/agreementService.ts

```diff
@@ -47,3 +147,222 @@ export const getLatestAgreement = async (
     })
     .at(0);
 };
+
+function isUpgradable(
+  descriptor: catalogApi.EServiceDescriptor,
+  agreement: agreementApi.Agreement,
+  descriptors: catalogApi.EServiceDescriptor[]
+): boolean {
+  return descriptors
+    .filter((d) => Number(d.version) > Number(descriptor.version))
+    .some(
+      (d) =>
+        (d.state === "PUBLISHED" || d.state === "SUSPENDED") &&
+        (agreement.state === "ACTIVE" || agreement.state === "SUSPENDED")
+    );
+}
```

Comment:
This function is already implemented and exported as `isAgreementUpgradable` in `bff/src/model/validators.ts` file

Reply:
Thanks a lot, refactoring it now :)